src/wing/CCPACSWings.cpp

```diff
@@ -141,18 +102,18 @@ bool CCPACSWings::HasWing(const std::string & uid) const
     return false;
 }
 
-void CCPACSWings::AddWing(CCPACSWing* wing)
+void CCPACSWings::Add(CCPACSWing* wing)
 {
-    // Check whether the same wing already exists if yes remove it before adding the new one
+    m_wings.push_back(unique_ptr<CCPACSWing>(wing));
+}
+
+void CCPACSWings::Remove(CCPACSWing* wing) {
     for (std::vector<unique_ptr<CCPACSWing> >::iterator it = m_wings.begin(); it != m_wings.end(); ++it) {
-        if ((*it)->GetUID() == wing->GetUID()) {
+        if ((*it).get() == wing) {
             m_wings.erase(it);
-            break;
+            return;
         }
     }
-
-    // Add the new wing to the wing list
-    m_wings.push_back(unique_ptr<CCPACSWing>(wing));
+    throw std::runtime_error("Wing not found");
```

Comment:
In the code I write I usually throw only std exceptions, because people seldomly care about the exception type, but the exception message. So I guess this line might just slipped in. I changed it to a TiglError.

However: TiglError should be changed to inherit from std::exception, otherwise it will not be caught in a:
´´´
try {
    func();
} catch(const std::exception& e) {
    // report error in e.what()
}
block. Thus any client code is required to explicitely catch CTiglError too to avoid being terminated instantly (unhandled exceptions terminate the process).
But this is subject to a subsequent PR, as a lot of changes to tigl.cpp are necessary than.